add docs about activities in Node

vkarpov15 opened the pull request:
## What does this PR do?

Added some learnings about using the Node SDK from working on [samples-node](https://github.com/vkarpov15/samples-node). There wasn't a doc on activities yet, so I added it

## Notes to reviewers

<!-- delete if n/a -->

bergundy commented on docs/node/activities.md:
```diff
@@ -0,0 +1,47 @@
+
+In Temporal, Activities are typically used to represent side effects, like making an HTTP request.
+Unlike [Workflows](/docs/node/logging#logging-from-workflows-and-activities), activities run in the standard Node.js environment, not an [isolate](https://www.npmjs.com/package/isolated-vm).
+So any code that needs to talk to the outside world, including importing modules, needs to be in an Activity.
```
Not true, you can import any module in WFs as long as it doesn't use the node or dom libs.

bergundy commented on docs/node/activities.md:
```diff
@@ -0,0 +1,47 @@
+# Activities
+
+In Temporal, Activities are typically used to represent side effects, like making an HTTP request.
+Unlike [Workflows](/docs/node/logging#logging-from-workflows-and-activities), activities run in the standard Node.js environment, not an [isolate](https://www.npmjs.com/package/isolated-vm).
```
Is there a better page we can link for workflows? Maybe the one about determinism?

bergundy commented on docs/node/activities.md:
```diff
@@ -0,0 +1,47 @@
+### Overview
+
+Below is a simple Activity that accepts a string parameter, appends a word to it, and returns the result.
+The Temporal Node SDK looks for any `.js` files in the `src/activities` directory, and automatically registers any exported functions as Activities.
```
Not accurate, in the package initializer, it's `.ts` in `src/activites` and the compiled code in `lib/activities` (configurable).

vkarpov15 replied:
Worth it to always use `lib/activities` rather than `src/activities`? If you're not using package initializer, either seems to work because of what [`activitiesPath` resolves to](https://github.com/temporalio/sdk-node/blob/9176b27270f7da77972ceb18c9c624f12e11ee93/packages/worker/src/worker.ts#L329)

bergundy reviewed (commented):
I'd add a section that explain that when you import an activity function from a WF it gets replaced with a stub that looks like this (psuedocode):

```
export function greet(...args) {
  return scheduleActivity('greet', args);
}
```

sw-yx reviewed (approved):
excellent, thank you

bergundy commented:
> I'd add a section that explain that when you import an activity function from a WF it gets replaced with a stub that looks like this (psuedocode):
> 
> ```
> export function greet(...args) {
>   return scheduleActivity('greet', args);
> }
> ```

I'm still missing this bit in the doc, but otherwise good to merge.
In the future we should save the user's TS/JS preference and use that for selecting the default shown tab.

flossypurse commented on docs/node/activities.md:
```diff
@@ -0,0 +1,68 @@
+# Activities
+
+In Temporal, Activities are typically used to represent side effects, like making an HTTP request.
```
"side effect" is actually the name of an API used to do some "non-deterministic" logic inside the Workflow code (at least in some of the other languages). Using it here as a way to describe an Activity is not ideal I think.

```suggestion
In Temporal, Activities are typically used to interact with external resources, like making an HTTP request.
```

flossypurse commented on docs/node/activities.md:
```diff
@@ -0,0 +1,68 @@
+# Activities
+
+In Temporal, Activities are typically used to represent side effects, like making an HTTP request.
+Unlike [Workflows](/docs/node/determinism), activities run in the standard Node.js environment, not an [isolate](https://www.npmjs.com/package/isolated-vm).
```
```suggestion
Unlike [Workflows](/docs/node/determinism), Activities execute in the standard Node.js environment, not an [isolate](https://www.npmjs.com/package/isolated-vm).
```

flossypurse commented on docs/node/activities.md:
```diff
@@ -0,0 +1,68 @@
+:::info
+
+Currently, you need to use the `@activities` prefix.
+Do **not** import your Activity using the `'../activities/greeter'` path, otherwise your Activity will run in the same V8 isolate as your Workflow and your Activity will be subject to the same restrictions as your Workflow.
```
```suggestion
Do **not** import your Activity using the `'../activities/greeter'` path, otherwise your Activity will execute in the same V8 isolate as your Workflow and your Activity will be subject to the same restrictions as your Workflow.
```

djmagee commented on docs/node/activities.md:
```diff
@@ -0,0 +1,68 @@
+:::info
+
+Currently, you need to use the `@activities` prefix.
+Do **not** import your Activity using the `'../activities/greeter'` path, otherwise your Activity will run in the same V8 isolate as your Workflow and your Activity will be subject to the same restrictions as your Workflow.
```
```suggestion
Do **not** import your Activity by using the `'../activities/greeter'` path; otherwise your Activity runs in the same V8 isolate as your Workflow, and your Activity is subject to the same restrictions as your Workflow.
```

vkarpov15 replied:
Thanks :+1:

vkarpov15 commented:
@bergundy I addressed your comments in https://github.com/temporalio/documentation/pull/473/commits/cf897bb84519c48f68f60a553077b6dacae4a425, let me know what you think :+1:

bergundy commented on docs/node/activities.md:
```diff
@@ -0,0 +1,72 @@
+</TabItem>
+</Tabs>
+
+Under the hood, the Temporal Node SDK uses [Webpack](https://webpack.js.org/) to bundle your Activities before running them.
```
Activities don't get bundled, only Workflows.

bergundy reviewed (commented):
LGTM apart from the part about bundling activities.
Feel free to merge after that's fixed.
